src/connectedvmware/azext_connectedvmware/custom.py

```diff
@@ -765,6 +768,146 @@ def get_hcrp_machine_id(
     return machine_id
 
 
+def reverseEndianLeftHalf(biosId: str):
+    """
+    Reverses the left half of the biosId.
+    This is required because the biosId format is different in HCRP Machine and InventoryItem.
+    Machine: 12345678-1234-1234-1234-123456789ABC
+    InventoryItem: 78563412-3412-3412-1234-123456789abc
+    """
+    def _rev(part: str):
+        return "".join(reversed([part[i:i + 2] for i in range(0, len(part), 2)]))
+    parts = biosId.split("-")
+    return "-".join(
+        [
+            _rev(parts[0]),
+            _rev(parts[1]),
+            _rev(parts[2]),
+            *parts[3:],
+        ]
+    )
+
+
+def link_vm_to_vcenter(
+    cmd,
+    client: VirtualMachineInstancesOperations,
+    vcenter,
+    rg_name=None,
+    resource_name=None,
+    no_wait=False,
+):
+    vcenter_id = vcenter
+    if rg_name is None and resource_name is not None:
+        raise RequiredArgumentMissingError(
+            "resource_group_name is required when machine_name is provided."
+        )
+    if not is_valid_resource_id(vcenter_id):
+        if not rg_name:
+            raise RequiredArgumentMissingError(
+                "Cannot determine vCenter ID. " +
+                "Resource group name is required when vCenter name is specified."
+            )
+        vcenter_id = get_resource_id(
+            cmd, rg_name, VMWARE_NAMESPACE, VCENTER_RESOURCE_TYPE, vcenter)
+    assert vcenter_id is not None
+    if resource_name is not None:
+        return create_vm(
+            cmd,
+            client,
+            resource_group_name=rg_name,
+            resource_name=resource_name,
+            vcenter=vcenter,
+            no_wait=no_wait,
+        )
+    
+    logger = get_logger(__name__)
+    arg_client = cf_resource_graph(cmd.cli_ctx)
+
+    vcenter_sub = vcenter_id.split("/")[2]
+    resources_client = get_resources_client(cmd.cli_ctx, vcenter_sub)
+    vcenter = resources_client.get_by_id(vcenter_id, VCENTER_KIND_GET_API_VERSION)
+    logger.info(f"Searching for machines in the vCenter {vcenter.name}...")
+
+    query = f"""
+Resources
+| where type =~ 'Microsoft.HybridCompute/machines'
+| where kind =~ "{vcenter.kind}" or isempty(kind)
+| extend p=parse_json(properties)
+| extend u = tolower(tostring(p['vmUuid']))
+| where isnotempty(u)
+| where location =~ '{vcenter.location}'
+| extend vmUuidRev = strcat(
+    substring(u, 6, 2), substring(u, 4, 2), substring(u, 2, 2), substring(u, 0, 2), '-',
+    substring(u, 11, 2), substring(u, 9, 2), '-',
+    substring(u, 16, 2), substring(u, 14, 2), '-',
+    substring(u, 19))
```

Comment:
we dont need to query all the machine resources, we can use resource name and RG scope if they are provided.

Reply:
Have made the changes